src/attribute.rs

```diff
@@ -741,11 +741,8 @@ impl<'n, 'f> Iterator for NtfsAttributesRaw<'n, 'f> {
 
         // It's a real attribute.
         let attribute = iter_try!(NtfsAttribute::new(self.file, self.items_range.start, None));
-        let length = usize::try_from(attribute.attribute_length()).ok()?;
-        if length == 0 {
-            return None;
-        }
-        self.items_range.start += length;
+        self.items_range.start += attribute.attribute_length() as usize;
```

Comment:
I get your point that `usize` should never be less that 32 bits and i agree but you removed the code that prevent the infinite loop 😢, because the problem wasn't really the u32 to usize conversion but more the fact that there is cases where length is always 0 and so the `next` iterator implemention is blocked in an infinite loop because never changes the state of the object.

This is why I propose just that instead.
```rust
let length = attribute.attribute_length() as usize;
if length == 0 {
    return None;
}
self.iterms_range.start += length
```

Reply:
Good catch! I actually removed it, because I meant to validate `attribute_length` to be at least `ATTRIBUTE_HEADER_SIZE` in the added `validate_attribute_length` code. Turns out that I just checked `remaining_length` but not `attribute_length`.

I just fixed this in e0e2fe524bc06a0d376cb562b13a2160ad4bac1f. By validating `attribute_length`, `length == 0` can no longer occur here and in all other cases where `NtfsAttribute::new` is used.